ksqldb-streams/src/main/java/io/confluent/ksql/execution/streams/materialization/ks/KsMaterializedTableIQv2.java

```diff
@@ -63,23 +63,13 @@
       final StateQueryResult<ValueAndTimestamp<GenericRow>>
           result = stateStore.getKafkaStreams().query(request);
 
-      final Map<Integer, QueryResult<ValueAndTimestamp<GenericRow>>> queryResultMap =
-          result.getPartitionResults();
-      if (queryResultMap.size() != 1) {
-        throw new MaterializationException("The query did not return exactly one partition result: "
-                + result.getPartitionResults());
-      }
-      if (!queryResultMap.containsKey(partition)) {
-        throw new MaterializationException(
-            "The query did not return the requested partition result: "
-                + result.getPartitionResults());
-      }
```

Comment:
These are checking for conditions that are in the IQv2 contract not to happen. Of course, there could be upstream bugs, but I don't know if we really need to check for them in our code.